### ISSUE TITLE
Unable to bin a jagged array

### ISSUE THREAD

**Issue by asnaylor** [FAST-HEP/fast-carpenter#67]
When trying to bin a `vector<float>` which is not a fixed size with the config below i get an index error message:
```yaml
stages:
- pre_cuts: fast_carpenter.Define
- cuts: fast_carpenter.CutFlow
- output: fast_carpenter.BinnedDataframe 

pre_cuts:
    variables:
        - s2: ms.s2Area_phd
        - multiple_scatter: (ms.nMultipleScatters > 0)
cuts: 
    selection: 
        Any: 
            - multiple_scatter
output:
    binning:
        - {in: s2}
```

Here is the traceback:
```python
Traceback (most recent call last):
  File "run.py", line 116, in <module>
    main()
  File "run.py", line 110, in main
    run_task()
  File "run.py", line 105, in run_task
    fast_carpenter_func()
  File "run.py", line 99, in fast_carpenter_func
    _, ret_val = run_carpenter(sequence, datasets, args)
  File "run.py", line 91, in run_carpenter
    ret_val = process.run(datasets, sequence)
  File "python_modules/atuproot/atuproot_main.py", line 59, in run
    result = self._run(loop)
  File "python_modules/atuproot/atuproot_main.py", line 121, in _run
    result = loop()
  File "python_modules/alphatwirl/datasetloop/loop.py", line 30, in __call__
    self.reader.read(dataset)
  File "python_modules/alphatwirl/datasetloop/reader.py", line 27, in read
    reader.read(dataset)
  File "python_modules/alphatwirl/loop/EventDatasetReader.py", line 66, in read
    runids = self.eventLoopRunner.run_multiple(eventLoops)
  File "python_modules/alphatwirl/loop/MPEventLoopRunner.py", line 93, in run_multiple
    return self.communicationChannel.put_multiple(eventLoops)
  File "python_modules/alphatwirl/concurrently/CommunicationChannel0.py", line 50, in put_multiple
    task_idx = self.put(t)
  File "python_modules/alphatwirl/concurrently/CommunicationChannel0.py", line 37, in put
    result = task(*args, **kwargs)
  File "python_modules/alphatwirl/loop/EventLoop.py", line 45, in __call__
    self.reader.event(event)
  File "python_modules/alphatwirl/loop/ReaderComposite.py", line 43, in event
    if reader.event(event) is False:
  File "python_modules/fast_carpenter/summary/binned_dataframe.py", line 189, in event
    data = chunk.tree.pandas.df(all_inputs)
  File "python_modules/fast_carpenter/masked_tree.py", line 27, in df
    df = self._owner.tree.pandas.df(*args, **kwargs)
  File "python_modules/fast_carpenter/tree_wrapper.py", line 70, in df
    df = self._owner.tree.pandas.df(*args, **kwargs)
  File "python_modules/uproot/_connect/_pandas.py", line 30, in df
    return self._tree.arrays(branches=branches, outputtype=pandas.DataFrame, namedecode=namedecode, entrystart=entrystart, entrystop=entrystop, flatten=flatten, flatname=flatname, awkwardlib=awkwardlib, cache=cache, basketcache=basketcache, keycache=keycache, executor=executor, blocking=blocking)
  File "python_modules/fast_carpenter/tree_wrapper.py", line 53, in arrays
    return self.tree.old_arrays(*args, **kwargs)
  File "python_modules/uproot/tree.py", line 484, in arrays
    return wait()
  File "python_modules/uproot/tree.py", line 468, in wait
    return uproot._connect._pandas.futures2df(futures, outputtype, entrystart, entrystop, flatten, flatname, awkward)
  File "python_modules/uproot/_connect/_pandas.py", line 192, in futures2df
    indexes = awkward.JaggedArray(starts, stops, awkward.numpy.empty(stops[-1], dtype=object)).tojagged(indexes).content
  File "python_modules/awkward/array/jagged.py", line 780, in tojagged
    content[good] = data[self.parents[good]]
IndexError: index 2325 is out of bounds for axis 1 with size 2319
```

**Comment by benkrikler** [comment]
Can you attach the versions of pandas, numpy, awkward array, and fast-carpenter that you're using?

**Comment by asnaylor** [comment]
```
pandas == 0.23.3
numpy == 1.14.2
awkward == 0.10.3
fast_carpenter == 0.13.1
```

**Comment by benkrikler** [comment]
Hmm those look a little old.  Can you try updating:
```
pip install fast-carpenter==0.13.2 pandas==0.25.1 awkward==0.12.8
```
and try again?

Be aware that if you update uproot to the latest version fast-carpenter has an issue: #68 so make sure you keep uproot  <= 3.9.0

**Comment by asnaylor** [comment]
I updated those packages you mentioned and i'm still having the same `IndexError` as before

**Comment by benkrikler** [comment]
I can reproduce this with LZ data now at least.

**Comment by benkrikler** [comment]
This was fixed by PR #71, and will be relased as v0.13.4, so I'm closing now.  @asnaylor please test if you get the chance!
